### PR TITLE
fix security bug

### PR DESCRIPTION
**Pull request by oshriromatrix**
Hi, in the file i found two line like this: 
 echo text($query) . "<br>\n"
This is a security bug, because the query should not be brought up to the user screen.

### CODE REVIEW

**Comment by bradymiller** [comment]
@oshriromatrix ,
While it is best to not display raw queries, note only the super admin can see this in theory and it does serve a purpose by showing the admin user what is getting deleted. The text() wrap stops html escaping, so not clear on the actual security vulnerability.

**Comment by oshriromatrix** [comment]
why administrator need to see query?
only developer need to see the query, when he doing a test for code.
if administrator want to see what he delete,we can send massage like:
the user ___ are delete from the system.
you never print your query out.

**Comment by bradymiller** [comment]
The issue here is that these are in generic functions and it will be difficult to print out sensical statements since the context has been lost. Could "hide" the query by simply doing a preg replace of:
'DELETE FROM' -> 'Removed'
'UPDATE' -> 'Revised'
(or something like that)

**Comment by oshriromatrix** [comment]
Hi,
I understand the superuser issue but ,
as you can see in debug mode I can catch this sql statement :   
image

This is a serious security breach . As I recall all statements are recorded at the log table in the DB.
Can we just leave that at this level  ?  or Can you think of another way I can fix out this issue.
I thought it would be best  maybe to put a global that handle that issue , like safe mode global or something like that . What do you think ?

**Comment by bradymiller** [comment]
We can figure something out here.
Just to help me out, what can you do from a security perspective if you catch the sql statement in debug mode?

**Comment by oshriromatrix** [comment]
Hi this is my  suggestion for security solution

File: library/globals.inc.php [hunk 1]
```diff
@@ -1573,7 +1573,12 @@
   // Security Tab
   //
   'Security' => array(
-
+     'safe_mode' => array(
+          xl('Safe mode'),
```
Comment (bradymiller) [inline library/globals.inc.php]: Lets change this to Mode - Do Not Show SQL Queries

File: library/globals.inc.php [hunk 2]
```diff
@@ -1573,7 +1573,12 @@
...
+          'bool',                           // data type
+          '0',                              // default
+          xl('Show to super user query and information of events')
```
Comment (bradymiller) [inline library/globals.inc.php]: Lets change this to:
Do not allow SQL queries to be outputted to screen.

File: library/globals.inc.php [hunk 3]
```diff
@@ -1573,7 +1573,12 @@
   // Security Tab
   //
   'Security' => array(
-
+     'safe_mode' => array(
```
Comment (bradymiller) [inline library/globals.inc.php]: change this sql_string_no_show_screen

**Comment by bradymiller** [comment]
That's a good compromise. Recommended some changes to make this more clear. After those are addressed, then will be ready to bring this into the codebase.

**Comment by oshriromatrix** [comment]
Hi this is my improvement by your recommended

**Comment by bradymiller** [comment]
Looks good and just brought this into the codebase. thanks! -brady

**Pull request closed without merging**